services/spar/src/Spar/Scim/User.hs

```diff
@@ -214,7 +212,14 @@ createValidScimUser
   :: forall m. (m ~ Scim.ScimHandler Spar)
   => ValidScimUser -> m Scim.StoredUser
 createValidScimUser (ValidScimUser user uref handl mbName) = do
-    -- This UserId will be used both for scim user in spar and for brig.
+    -- FUTUREWORK: The @hscim@ library checks that the handle is not taken before 'create' is
+    -- even called. However, it does that in an inefficient manner. We should remove the check
+    -- from @hscim@ and do it here instead.
+
+    -- Check that the UserRef is not taken.
+    whenM (isJust <$> lift (wrapMonadClient (Data.getUser uref))) $
+        throwError Scim.conflict {Scim.detail = Just "externalId is already taken"}
```

Comment:
Is this something that needs to be fixed independently of re-auth for scim token creation?

Reply:
It's already in develop (I don't know how it ended up there, actually).